addons/survey/static/src/js/survey_form.js

```diff
@@ -981,6 +991,34 @@ publicWidget.registry.SurveyFormWidget = publicWidget.Widget.extend({
         }
     },
 
+    /**
+    * This method check if the current tab is the master tab at the bus level.
+    * If not, the survey could not receive next question notification anymore from session manager.
+    * We then ask the participant to close all other tabs on the same hostname before letting them continue.
+    *
+    * @private
+    */
+    _checkIsMasterTab: function () {
+        var isMasterTab = this.call('bus_service', 'isMasterTab');
+        var $errorModal = this.$('#AJAXErrorModal');
+        if (isMasterTab) {
+            // Force reload the page when survey is ready to be followed, to force restart long polling
+            if (this.shouldReloadMasterTab) {
+                window.location.reload();
+            }
+           return true;
+        } else if ($errorModal[0].style.display !== 'block'){
```

Comment:
Several things here:
- Wow...
- Missing translations (good luck with PO files)
- Could build the blocks with $('<p>', {text: XXX}) and append spans instead of using a big ".html()" as this is an XSS injection point (I can elaborate on that if needed)
- And finally... wow...

Reply:
much wow :D